### DISCUSSION TITLE
Understanding Pre-Rendering For A Blog

### DISCUSSION THREAD

**Discussion by wazcov** [vercel/next.js#32189] [Help]
I have a website that I've just built on NextJS. It consists of a tool where users can find information on a particular topic, from a database, and a blog alongside it. 

Now currently I do not implement any of the `getServerSideProps` or `getStaticProps` methods. I believe this means that my site is not prerendered, and this means it's generating the information on every request. (Correct?)

Now SEO is really important for this business, and I believe that currently Google will not correctly index my dynamic routes, nor find the content on the pages. (Correct?)

If my assumptions are correct, which method am I best going with to help SEO - SSR or Static Generation? Note there's some external API calls but it doesn't matter if they are an hour or so out of date. All the other data comes from my own database (via the internal next API), which I wish to have updated and new blog posts appear without me having to rebuild the app.

Many thanks

**Comment by delbaoliveira** [comment]
Hi @wazcov,

> Now currently I do not implement any of the getServerSideProps or getStaticProps methods. I believe this means that my site is not prerendered, and this means it's generating the information on every request. (Correct?)

If you're not using `getStaticProps()`  or  `getServerSideProps()`, then it sounds like you might be fetching your data on the client-side using `useEffect()` or a data fetching library such as [useSWR](https://swr.vercel.app/)? 

Any data you fetch on the client-side might not be indexed as efficiently by search engines. 

> which I wish to have updated and new blog posts appear without me having to rebuild the app.

For this kind of app, the simpler solution would be to use `getStaticProps()` to fetch from your two sources. You can use [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) to periodically refetch your data and rebuild the page in the background.  

You can use `getServerSideProps()` if you'd like, but I'd advise using cache headers for performance. Here's an [example](https://github.com/vercel/next.js/blob/canary/examples/ssr-caching) if you're hosting on Vercel ([docs](https://vercel.com/docs/concepts/edge-network/caching#stale-while-revalidate)). 

> Which method am I best going with to help SEO - SSR or Static Generation?

Both methods work well in comparison to client-side rendering because the content will be pre-rendered ahead of time and sent as part of the initial HTML. This makes it easier for search engines to index your pages. 

> Now SEO is really important for this business, and I believe that currently, Google will not correctly index my dynamic routes, nor find the content on the pages. (Correct?)

If you use `getStaticPaths()` in combination with a dynamic route, Next.js will generate a page for each of those routes which is good for SEO. You can then link to those pages from your blog index page. 

--- 
We have an [SEO course](https://nextjs.org/learn/seo/introduction-to-seo) on Next.js Learn that dives deeper into the topic. If you have the time, may I recommend this section on [rendering strategies](https://nextjs.org/learn/seo/rendering-and-ranking/rendering-strategies) and this section on how [web crawlers work](https://nextjs.org/learn/seo/introduction-to-seo/webcrawlers).
